java/registry/src/main/java/dev/sunbirdrc/registry/service/FileStorageService.java

```diff
@@ -139,4 +150,46 @@ public ComponentHealthInfo getHealthInfo() {
 			return new ComponentHealthInfo(getServiceName(), false, CONNECTION_FAILURE, e.getMessage());
 		}
 	}
+
+	public DocumentsResponse updateFiles(MultipartFile[] files, String requestedURI) {
+		String objectPath = getDirectoryPath(requestedURI);
+
+		DocumentsResponse documentsResponse = new DocumentsResponse();
+		for (MultipartFile file : files) {
+			String fileName = getFileName(file.getOriginalFilename());
+			try {
+				String objectName = objectPath;
+
+				boolean objectExists = minioClient.statObject(
+						StatObjectArgs.builder().bucket(bucketName).object(objectName).build()
+				) != null;
+
+				if (objectExists) {
+					save(file.getInputStream(),objectName);
+					documentsResponse.addDocumentLocation(objectName);
+					return documentsResponse;
+				}
+
+				for (MultipartFile file1 : files) {
```

Comment:
I would say lets not save the file if it is not present. Lets only update the file if present or else throw appropriate error